Remove repo from suggestions if invalid

wx-03 opened the pull request:
### Summary:

Fixes #397 

#### Type of change:

- 🐛 Bug Fix

### Changes Made:

- Removed a repo from suggestions if it is not a valid repo
- Edited spy object in testing to include new function

### Proposed Commit Message:

```
Remove repo from suggestions if invalid

WATcher does not remove a repo from suggestions if it is not valid. 
This results in invalid repositories shown under suggestions if 
the user has previously entered one. 

Let's remove invalid repos from suggestions since users do not need 
to choose them.
```

<details><summary>
<h3>Checklist:</h3>
</summary>

- [ ] I have tested my changes thoroughly.
- [ ] I have created tests for any new code files created in this PR or provided a link to a issue/PR that addresses this.
- [ ] I have added or modified code comments to improve code readability where necessary.
- [ ] I have updated the project's documentation as necessary.

</details>

Eclipse-Dominator commented on src/app/core/services/repo-url-cache.service.ts:
```diff
@@ -23,6 +23,12 @@ export class RepoUrlCacheService {
   }
 
+  removeFromSuggestions(repo: string): void {
+    const repoIndex = this.suggestions.indexOf(repo);
+    this.suggestions.splice(repoIndex, 1);
```
I'm not sure if we are always guaranteed to find a result (without manual modification to the cache),
but to ensure safety, just return immediately if the repo is not found in the cache.

<img width="398" alt="image" src="https://github.com/user-attachments/assets/19f95836-b3c3-44bf-9c4e-71e4f2b8ca0e" />


This is because splice can work on negative numbers as seen above.
Thus, it is good if we just exit the function straight away if the repo didnt exist in cache.

Eclipse-Dominator replied:
Might be good to include this as a test case for this file.

Eclipse-Dominator reviewed (changes requested):
Functionality works but might be unsafe.

wx-03 commented:
> Good work! Functionality works but might be unsafe (see comments below)

@Eclipse-Dominator Thanks for the review, I've updated the function and added some tests to ensure that it runs correctly.

Eclipse-Dominator reviewed (changes requested):
Apologies, I missed this in the previous PR.

Eclipse-Dominator reviewed (approved):
The NG espression changed error is a bug related to the hidden group component that wasn't fixed. Thus this PR is functioning as expected

LGTM
